### PR TITLE
[BD-46] feat: add SCSS linter and fix SCSS files

### PR DESCRIPTION
**Pull request by monteri**
Linter for SCSS:
- linter config with rules for Paragon
- precommit linting for `paragon` and `paragon/www` `*.scss` files
- file fixes according to new rules

JIRA: https://openedx.atlassian.net/browse/PAR-794

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +46,7 @@
     "font-awesome": "^4.7.0",
     "lodash.uniqby": "^4.7.0",
     "mailto-link": "^1.0.0",
-    "prop-types": "^15.8.1",
+    "prop-types": "^15.7.2",
```
Comment (viktorrusakov) [inline package.json:49]: Why do we downgrade here?

**Comment by viktorrusakov** [comment]
I also have the same concern with your `package-lock` as in #1248, see [my comment](https://github.com/openedx/paragon/pull/1248) there, can you please double check that your `package-lock` is correct?

File: src/Breadcrumb/_variables.scss [hunk 2]
```diff
@@ -17,6 +16,6 @@ $breadcrumb-inverse-active:         $light-500 !default;
 $breadcrumb-inverse-spacer:         $light-700 !default;
 $breadcrumb-color:                  $primary-500 !default;
 $breadcrumb-inverse-color:          $white !default;
-$breadcrumb-divider:                quote("/") !default;
+$breadcrumb-divider:                "/";
```
Comment (viktorrusakov) [inline src/Breadcrumb/_variables.scss:19]: should we be removing `!default` here?

File: package.json (hunk 1 above)
Comment (monteri) [reply package.json:49]: Npm `8.7` changes automatically for some reason. I reverted back

File: src/Breadcrumb/_variables.scss (hunk 2 above)
Comment (monteri) [reply src/Breadcrumb/_variables.scss:19]: Yes! Adding `!default` here, thank you.

File: .stylelintrc.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }],
+    "property-no-vendor-prefix": [true, {
+      "ignoreProperties": ["animation", "filter"]
+    }],
+    "value-no-vendor-prefix": [true, {
+      "ignoreValues": ["fill-available"]
+    }],
+    "number-leading-zero": "never",
+    "no-descending-specificity": null,
+    "selector-class-pattern": "^(pgn__|pgn-doc__)?([a-z][a-z0-9]*)(-{1,2}[a-z0-9]+)*$",
```
[the remark this answers is not in the document]
Comment (monteri) [reply .stylelintrc.json]: I added and tested BEM regex instead of previous regex version.

**Review by adamstankiewicz — DISMISSED** [review]

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@monteri 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1243&entry.1484655318=https://github.com/monteri&entry.752974735=2022-04-20+13:25&entry.1917517419=2022-05-13+10:08&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**